File: C Sharp/Object Validator/ValidationResult.cs
namespace Quicksilver.ObjectValidator;
public class ValidationResult
{
    public readonly bool passed;
    public readonly ISet<string> failedFields;
    public readonly ICollection<ValidationFailure> failures;

    internal ValidationResult(bool passed, ISet<string> failedFields, ICollection<ValidationFailure> failures)
    {
        this.passed = passed;
        this.failedFields = failedFields;
        this.failures = failures;
    }
}

File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/ValidationResult.java
package com.quicksilver.objectvalidator;

import java.util.Collection;
import java.util.Set;

public class ValidationResult {
    public final boolean passed;
    public final Set<String> failedFields;
    public final Collection<ValidationFailure> failures;

    public ValidationResult(boolean passed, Set<String> failedFields, Collection<ValidationFailure> failures) {
        this.passed = passed;
        this.failedFields = failedFields;
        this.failures = failures;
    }
}
